althea_kernel_interface/src/interface_tools.rs

```diff
@@ -101,6 +101,41 @@ impl dyn KernelInterface {
             static ref RE: Regex = Regex::new(r"((\d){1,3}\.){3}(\d){1,3}/(\d){1,3}")
                 .expect("Unable to compile regular expression");
         }
+
+        let mut ret = Vec::new();
+        for line in stdout.lines() {
+            let cap = RE.captures(line);
+            // we captured something on this line
+            if let Some(cap) = cap {
+                // flatten drops the 'none' values in this array
+                for ip_cap in cap.iter().flatten() {
+                    let mut split = ip_cap.as_str().split('/');
+                    let ip_str = split.next();
+                    let netmask = split.next();
+                    if let (Some(ip_str), Some(netmask)) = (ip_str, netmask) {
+                        if let (Ok(parsed_ip), Ok(parsed_netmask)) =
+                            (ip_str.parse(), netmask.parse())
+                        {
+                            ret.push((parsed_ip, parsed_netmask));
+                        }
+                    }
+                }
+            }
+        }
+
+        Ok(ret)
+    }
+
+    /// Gets all the IPv6 addresses from an interface and returns the address and it's netmask
+    /// as a tuple.
+    pub fn get_ipv6_from_iface(&self, name: &str) -> Result<Vec<(IpAddr, u8)>, Error> {
+        let output = self.run_command("ip", &["address", "show", "dev", name])?;
```

Comment:
you should use the ipv6 and ipv4 types instead of ipaddr that way there can't possibly be any confusion about what type of ip we're getting back.

Reply:
fixed